src/main/io/osd.c

```diff
@@ -400,6 +403,18 @@ static void osdFormatMessage(char *buff, size_t size, const char *message)
     if (message) {
         memcpy(buff, message, strlen(message));
     }
+
+    // Save warning into pilotConfig->warning, used for DJI OSD
+    // stored into another field for saving original pilot name
+    if (osdWarnGetState(OSD_WARNING_DJI)) {
+        char warning[OSD_FORMAT_MESSAGE_BUFFER_SIZE] = " ";
+        osdConfigMutable()->item_pos[OSD_CRAFT_NAME] = osdConfig()->item_pos[OSD_WARNINGS]; // Change position of craft name
+        if (message) {
+            memcpy(warning, message, OSD_WARNINGS_MAX_SIZE);
+        }
+        memcpy(pilotConfigMutable()->warning, warning, OSD_WARNINGS_MAX_SIZE);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Does not check for buffer overflows when copying to destination (CWE-120).  Make sure destination can always hold the source data. ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5623733)